source/TestRunner/Program.cs

```diff
@@ -0,0 +1,130 @@
+﻿using BannerlordSystemTestingLibrary;
+using SuperWebSocket;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace TestRunner
+{
+    /// <summary>
+    /// Runner that works similar to running a test but as an application
+    /// </summary>
+    static class TestRunner
+    {
+        [DllImport("Kernel32")]
+        private static extern bool SetConsoleCtrlHandler(EventHandler handler, bool add);
+        private delegate bool EventHandler(CtrlType sig);
+        static EventHandler _handler;
+
+        static GameInstance host = new GameInstance("/singleplayer /server _MODULES_*Native*SandBoxCore*CustomBattle*SandBox*StoryMode*Coop*_MODULES_");
+        static GameInstance client = new GameInstance("/singleplayer /client _MODULES_*Native*SandBoxCore*CustomBattle*SandBox*StoryMode*Coop*_MODULES_");
+        //static GameInstance client2 = new GameInstance("/singleplayer /client _MODULES_*Native*SandBoxCore*CustomBattle*SandBox*StoryMode*Coop*_MODULES_");
+
+        static List<GameInstance> instances = new List<GameInstance>
+            {
+                //host,
+                //client,
+            };
+
+        static TestEnvironment environment;
+
+        static void Main(string[] args)
+        {
+            environment = new TestEnvironment(instances);
+
+            // Some biolerplate to react to close window event
+            _handler += new EventHandler(Handler);
+            SetConsoleCtrlHandler(_handler, true);
+            
+
+            environment.OnRegistrationFinished += (instance) =>
+            {
+                instance.OnGameStateChanged += (state) => {
+                    if (state == GameStates.MainMenuReadyState)
+                    {
+                        instance.SendCommand("StartCoop");
```

Comment:
From what i've gather so far, this invokes `TestingCommands.StartCoop` on the instance that is being tested? Is the string `StartCoop` gathered from the MethodInfo of the TestingCommands class?

Reply:
That is correct.